### PR TITLE
[ci] Enable pylint for tests/python/ci

### PR DESCRIPTION
**Pull request by driazati**
This fixes up the pylint issues as part of #11414 for the CI tests

Thanks for contributing to TVM!   Please refer to guideline https://tvm.apache.org/docs/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/apache/incubator-tvm/blob/master/CONTRIBUTORS.md#reviewers) by @ them in the pull request thread.

cc @Mousius @areusch

### CODE REVIEW

**Review by areusch — APPROVED** [review]

**Comment by areusch** [comment]
thanks @driazati, please check off the relevant items in the GH issue when you merge

File: tests/python/ci/test_ci.py [hunk 1]
```diff
@@ -49,7 +38,7 @@ def run(pr_body, requested_reviewers, existing_review_users, expected_reviewers)
         git.run("remote", "add", "origin", "https://github.com/apache/tvm.git")
         reviews = [{"user": {"login": r}} for r in existing_review_users]
         requested_reviewers = [{"login": r} for r in requested_reviewers]
-        proc = subprocess.run(
+        proc = subprocess.run(  # pylint: disable=subprocess-run-check
```
Comment (Mousius) [inline tests/python/ci/test_ci.py:41]: Looking at the `subprocess-run-check` linter rule, shouldn't we pass `check` explicitly to resolve it?

File: tests/python/ci/test_ci.py [hunk 2]
```diff
@@ -703,27 +713,31 @@ def run(type, data, check):
...
             "title": "[something] A title",
             "number": 1234,
             "user": {
                 "login": "person5",
             },
             "labels": [{"name": "something2"}],
+            # pylint: disable=line-too-long
             "body": textwrap.dedent(
                 """
                 `mold` and `lld` can be a much faster alternative to `ld` from gcc. We should modify our CMakeLists.txt to detect and use these when possible. cc @person1
```
Comment (Mousius) [inline tests/python/ci/test_ci.py:727]: Can we not have a line per sentence here?

File: tests/python/ci/test_ci.py [hunk 3]
```diff
@@ -342,9 +341,12 @@ def run(files, should_skip):
 
 
 def test_ping_reviewers(tmpdir_factory):
+    """
+    Test that reviewers are messaged after a time period of inactivity
+    """
     reviewers_script = REPO_ROOT / "tests" / "scripts" / "ping_reviewers.py"
 
-    def run(pr, check):
+    def run(pr, check):  # pylint: disable=invalid-name
```
Comment (Mousius) [inline tests/python/ci/test_ci.py:349]: Wouldn't it be better to fix the function name than ignore the linter warning?

**Review by Mousius — CHANGES_REQUESTED** [review]
It appears we're disabling a lot of the rules in this PR, could you help me understand the rationale a bit @driazati? 😸

File: tests/python/ci/test_ci.py (hunk 1 above)
Comment (driazati) [reply tests/python/ci/test_ci.py:41]: It's checked manually to add some extra error info

File: tests/python/ci/test_ci.py (hunk 3 above)
Comment (driazati) [reply tests/python/ci/test_ci.py:349]: the issue is on `pr` (says its too short), but that is what it is and I don't think the linter is correct here

File: tests/python/ci/test_ci.py (hunk 2 above)
Comment (driazati) [reply tests/python/ci/test_ci.py:727]: we could but I would argue that makes this harder to read and isn't the idiomatic way to include a long message, hence the linter disable

Comment (areusch) [reply tests/python/ci/test_ci.py:727]: i think we should just be consistent across whatever we do on line 734 as well

File: tests/python/ci/test_ci.py (hunk 1 above)
Comment (Mousius) [reply tests/python/ci/test_ci.py:41]: If you set `check=False`, that continues to work as far as I can tell?

File: tests/python/ci/test_ci.py (hunk 2 above)
Comment (Mousius) [reply tests/python/ci/test_ci.py:727]: Agree with @areusch we should be consistent. The reason I'd err on the side of splitting is so it's easier to read on as many peoples screens as possible, without side scrolling or the editor trying to line wrap it.

File: tests/python/ci/test_ci.py (hunk 3 above)
Comment (Mousius) [reply tests/python/ci/test_ci.py:349]: I don't see a reason to not use `pull_request` here which is more immediately obvious to the reader?

File: tests/python/ci/test_ci.py (hunk 1 above)
Comment (driazati) [reply tests/python/ci/test_ci.py:41]: That still trips pylint as far as I can tell, it wants to see `check=True` only

Comment (Mousius) [reply tests/python/ci/test_ci.py:76]: Can you check again, I tried it locally and it worked 🤔

File: tests/python/ci/test_utils.py [hunk 4]
```diff
@@ -14,7 +14,29 @@
...
+    """
+
+    def __init__(self, cwd):
+        self.cwd = cwd
+
+    def run(self, *args, **kwargs):
+        # the returncode is checked manually, ignore pylint
+        proc = subprocess.run(  # pylint: disable=subprocess-run-check
+            ["git"] + list(args), encoding="utf-8", cwd=self.cwd, **kwargs
+        )
```
Comment (Mousius) [inline tests/python/ci/test_utils.py:38]:
```suggestion
    def run(self, *args, **kwargs):
        """Runs a git command in the temporary directory"""
        proc = subprocess.run(
            ["git"] + list(args),
            encoding="utf-8",
            check=False,
            cwd=self.cwd,
            **kwargs,
        )
```

File: tests/python/ci/test_mergebot.py [hunk 5]
```diff
@@ -165,7 +158,8 @@ def test_mergebot(tmpdir_factory, number, filename, expected, comment, user, det
     }
     collaborators = []
 
-    proc = subprocess.run(
+    # return code is checked manually, so disable pylint
+    proc = subprocess.run(  # pylint: disable=subprocess-run-check
```
Comment (Mousius) [inline tests/python/ci/test_mergebot.py:162]: We can use `check=False` here as well I think? I ran the tests and linter locally to check everything still works.

**Review by Mousius — CHANGES_REQUESTED** [review]
I think there's just two outstanding things here @driazati, ping me once they're in so we can unblock #11714 😸

**Comment by driazati** [comment]
@Mousius this is ready for another look

**Review by Mousius — APPROVED** [review]
Looks good @driazati, just has some merge cornflakes this morning - happy for you to merge once resolved.

**Pull request merged**